python/ql/lib/semmle/python/frameworks/Stdlib.qll

```diff
@@ -4396,6 +4396,145 @@ private module StdlibPrivate {
       preservesValue = true
     }
   }
+
+  // ---------------------------------------------------------------------------
+  // asyncio
+  // ---------------------------------------------------------------------------
+  /** Gets a reference to the `asyncio` module. */
+  API::Node asyncio() { result = API::moduleImport("asyncio") }
+
+  /** Provides models for the `asyncio` module. */
+  module AsyncIO {
+    /**
+     * A call to the `asyncio.create_subprocess_exec` function (also accessible via the `subprocess` module of `asyncio`)
+     *
+     * See https://docs.python.org/3/library/asyncio-subprocess.html#creating-subprocesses
+     */
+    private class CreateSubprocessExec extends SystemCommandExecution::Range,
+      FileSystemAccess::Range, DataFlow::CallCfgNode
+    {
+      CreateSubprocessExec() {
+        exists(string name |
+          name = "create_subprocess_exec" and
+          (
+            this = asyncio().getMember(name).getACall()
+            or
+            this = asyncio().getMember("subprocess").getMember(name).getACall()
+          )
+        )
+      }
+
+      override DataFlow::Node getCommand() {
+        result = this.getArg(0)
+        or
+        result = this.getArgByName("program")
+      }
+
+      override DataFlow::Node getAPathArgument() { result = this.getCommand() }
+
+      override predicate isShellInterpreted(DataFlow::Node arg) {
+        none() // this is a safe API.
+      }
+    }
+  }
```

Comment:
`API::CallNode` is a bit more powerful than `DataFlow::CallCfgNode`, so we've started using that when we can instead :blush: (that is what provides the `getParameter` member-predicate)

```suggestion
    private class CreateSubprocessExec extends SystemCommandExecution::Range,
      FileSystemAccess::Range, API::CallNode
    {
      CreateSubprocessExec() {
        this = API::moduleImport("asyncio").getMember("create_subprocess_exec").getACall()
        or
        this =
          API::moduleImport("asyncio")
              .getMember("subprocess")
              .getMember("create_subprocess_exec")
              .getACall()
      }

      override DataFlow::Node getCommand() { result = this.getParameter(0, "program").asSink() }

      override DataFlow::Node getAPathArgument() { result = this.getCommand() }

      override predicate isShellInterpreted(DataFlow::Node arg) {
        none() // this is a safe API.
      }
    }
  }
```